src/test/java/it/pagopa/pn/bff/utils/notificationDetail/NotificationDetailUtilityTest.java

```diff
@@ -0,0 +1,297 @@
+package it.pagopa.pn.bff.utils.notificationDetail;
+
+import it.pagopa.pn.bff.generated.openapi.msclient.delivery_recipient.model.FullReceivedNotificationV23;
+import it.pagopa.pn.bff.generated.openapi.msclient.delivery_recipient.model.NotificationStatusHistoryElement;
+import it.pagopa.pn.bff.generated.openapi.msclient.delivery_recipient.model.TimelineElementCategoryV23;
+import it.pagopa.pn.bff.generated.openapi.msclient.delivery_recipient.model.TimelineElementV23;
+import it.pagopa.pn.bff.generated.openapi.server.v1.dto.*;
+import it.pagopa.pn.bff.mapper.NotificationDetailMapper;
+import it.pagopa.pn.bff.mocks.NotificationDetailMock;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.BeanUtils;
+
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class NotificationDetailUtilityTest {
+    NotificationDetailMock notificationDetailMock = new NotificationDetailMock();
+
+    @Test
+    void populateOtherDocumentsTest() {
+        FullReceivedNotificationV23 notificationDTOMultiRecipient = notificationDetailMock.notificationMultiRecipientMock();
+        FullReceivedNotificationV23 notificationDTO = notificationDetailMock.getOneRecipientNotification();
+
+        FullReceivedNotificationV23 noAARNotification = new FullReceivedNotificationV23();
+        BeanUtils.copyProperties(notificationDTOMultiRecipient, noAARNotification);
+        noAARNotification.setTimeline(noAARNotification.getTimeline().stream().filter(
+                t -> t.getCategory() != TimelineElementCategoryV23.AAR_GENERATION
+        ).collect(Collectors.toList()));
+
+        BffFullNotificationV1 calculatedParsedNotification = NotificationDetailMapper.modelMapper.mapNotificationDetail(noAARNotification);
+
+        assert calculatedParsedNotification.getOtherDocuments().isEmpty();
+
+        ArrayList<TimelineElementV23> AARTimelineElements = notificationDTO.getTimeline().stream().filter(
+                t -> t.getCategory() == TimelineElementCategoryV23.AAR_GENERATION
+        ).collect(Collectors.toCollection(ArrayList::new));
+
+        calculatedParsedNotification = NotificationDetailMapper.modelMapper.mapNotificationDetail(notificationDTO);
+
+        assert calculatedParsedNotification.getOtherDocuments().size() == 1;
+
+        assert calculatedParsedNotification.getOtherDocuments().get(0).equals(
```

Comment:
for this condition and below, separate the assert from the object that is asserted